src/outcome/utils/cache.py

```diff
@@ -118,16 +125,34 @@ def configure_cache_region(cache_region: CacheRegion, settings: Dict[str, Any],
 
 class TTLBackend(CacheBackend):
     def __init__(self, arguments):
-        self.cache = TTLCache(**arguments)
+        self.persisted_cache_path = arguments.get('path')
+        if self.persisted_cache_path:
+            try:
+                with open(self.persisted_cache_path, 'rb') as f:
+                    self.cache = pickle.load(f)  # noqa: S301
```

Comment:
What if the values in `arguments` have changed since the cache was pickled?

Reply:
I'm not sure of how we want to check this ?

Reply:
I'd say we need to update the unpickled cache with the new arguments

Reply:
I can't manage to set these attributes once the TTLCache exists - I get `AttributeError: can't set attribute`, I guess because they're TTLCache properties ?

Reply:
```suggestion
          _cache = TTLCache(**arguments)
          ...
                with open(self.persisted_cache_path, 'rb') as f:
                    self.cache = pickle.load(f)  # noqa: S301
                    for arg_key in arguments.keys():
                        if getattr(self.cache, arg_key) != getattr(_cache, arg_key):
                            self.cache = _cache
                            break
```

I could try something like that ?